### PR TITLE
TE-640 Add fullscreen option to modal

### PR DESCRIPTION
**Pull request by marcdavi-es**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-640)

### What **one** thing does this PR do?
Add fullscreen option to modal

kapture 2018-07-19 at 18 15 56

### CODE REVIEW

File: src/styles/semantic/themes/livingstone/modules/modal.overrides [hunk 1]
```diff
@@ -36,6 +36,13 @@
 /* Mobile Positions */
 
 /* Responsive Widths */
+.ui.fullscreen.modal {
+  bottom: 0;
+  margin: 0;
+  margin-bottom: @fullScreenMarginBottom;
+  position: absolute;
+  top: 0;
+}
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/modal.overrides]: I think we need to account for content expanding outside of the modal container,

Add the following css to the `.content` of the modal
```
    overflow: auto;
    max-height: 100vh;
```

File: src/styles/semantic/themes/livingstone/modules/modal.overrides [hunk 2]
```diff
@@ -36,6 +36,13 @@
 /* Mobile Positions */
 
 /* Responsive Widths */
+.ui.fullscreen.modal {
+  bottom: 0;
+  margin: 0;
+  margin-bottom: @fullScreenMarginBottom;
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/modal.overrides]: You can just use the shorthand `margin: 0 0 @ fullScreenMarginBottom;`

File: src/styles/semantic/themes/livingstone/modules/modal.variables [hunk 3]
```diff
@@ -25,6 +25,9 @@
 /* Mobile Positions */
 
 /* Responsive Widths */
+@fullScreenWidth: 100%;
+// Fixes the bug I describe at https://stackoverflow.com/questions/41453780/semantic-react-modal-weird-height-behavior/51426377#51426377
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/modal.variables]: The fix looks good, but an alternative approach could be to set a max height of `max-height: calc(100% - 1px);` or `max-height: calc(100vh - 1px);`

Comment (marcdavi-es) [reply src/styles/semantic/themes/livingstone/modules/modal.variables]: Yep on reflection I think `height: calc(100% - 1px);` is the least hacky of the available hacks :)

Comment (stephenkr) [reply src/styles/semantic/themes/livingstone/modules/modal.variables]: Yep that looks good 👍

**Pull request merged**